### PR TITLE
[improvement](timezone) support mixed uppper-lower case of timezone names

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

File: be/src/util/timezone_utils.cpp [hunk 1]
```diff
@@ -21,13 +21,43 @@
...
 
 RE2 TimezoneUtils::time_zone_offset_format_reg("^[+-]{1}\\d{2}\\:\\d{2}$");
+DorisCallOnce<Status> TimezoneUtils::load_timezone_once_;
+std::unordered_map<std::string, std::string> TimezoneUtils::timezone_names_map_;
 
 const std::string TimezoneUtils::default_time_zone = "+08:00";
 
+void TimezoneUtils::_load_timezone_names() {
+    std::string path;
+    const char* tzdir = "/usr/share/zoneinfo";
```
Comment (yiguolei) [inline be/src/util/timezone_utils.cpp:38]: maybe it is not right. For exmple, in macos, the dir is not right. Or in some other linux distributions.
Could we add a static map in doris's code?

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**